### PR TITLE
fix: no-wrap parameter

### LINKED ISSUE #17: --no-wrap does not work
Always puts in a directory.

### PR DESCRIPTION
**Pull request by alanshaw**
fixes #17

### CODE REVIEW

File: index.js [hunk 1]
```diff
@@ -29,8 +29,11 @@ export async function upload (firstPath, opts) {
     }
   }
   spinner.start('Storing')
-  // @ts-ignore
-  const root = await client.uploadDirectory(files, {
+  /** @type {(o?: import('@web3-storage/w3up-client/src/types').UploadOptions) => Promise<import('@web3-storage/w3up-client/src/types').AnyLink>} */
+  const uploadFn = files.length === 1 && opts['no-wrap']
```
Comment (olizilla) [inline index.js:33]: i think files.length would be > 1 if you added
```
foo
├── alan.txt
└── bar.txt
```
is it  that directories are always wrapped so the `foo` is preserved?

we should warn the user that --no-wrap is ignored in that case, or error

Comment (alanshaw) [reply index.js:33]: Ah, `files-from-path` is relative to working directory, not common path...

**Comment by jchris** [comment]
is this related to our ability to hide the directory selection stuff from users? https://github.com/web3-storage/w3ui/issues/139

**Comment by olizilla** [comment]
This is "when encoding a file to unixfs; should we create a directory for you with that single thing in it to preserve your things filename in the dag". If we don't then you are gonna get a CID for the bytes of your file, but your filename is not captured.

**Review by olizilla — APPROVED** [review]
👨‍🎤 ✨

**Pull request merged**